### DISCUSSION TITLE
How can I change variable?

### DISCUSSION THREAD

**Discussion by MisterMakerNL** [wg-easy/wg-easy#1538] [Q&A]
First off great piece of software works awesome! Thanks for making it!
I tried testing it with my IP address, but now I want to set it to a dyndns url.
But I cannot get it to change the url.
I don't understand what too do here:
https://github.com/wg-easy/wg-easy?tab=readme-ov-file#options
Could someone give an example on how to change a variable?

**Comment by shalafi99** [comment]
How are you creating the wg-easy container?

docker run, docker composer or some other way?

**Reply by MisterMakerNL** [reply]
I followed the guide. So yes in a container.

**Reply by shalafi99** [reply]
It turns out the guide has instructions for doing it with docker run OR docker composer.
Can you share the command or file you used for the container creation?
(Mask the password hash)

**Reply by MisterMakerNL** [reply]
I tried this:
> root@VPN:~# docker run  -e WG_HOST='vpn.********.com'
> "docker run" requires at least 1 argument.
> See 'docker run --help'.
> 
> Usage:  docker run [OPTIONS] IMAGE [COMMAND] [ARG...]
> 
> Create and run a new container from an image
> root@VPN:~#

**Reply by shalafi99** [reply]
I might have misunderstood what you are trying to achieve while I read the original post.
You want to change that environment variable for the currently existing/running wg-easy container?

`docker run -e WG_HOST='vpn.********.com'`
Were this the command to *create* a new container, it is missing the image name at the end (ghcr.io/wg-easy/wg-easy in this case)

But if it is an existing container, you need to recreate it and this time you would use a different value for WG_HOST in relation to the first time - cannot change variables for a running docker container.

**Reply by MisterMakerNL** [reply]
Ok so I am missing some vital info :)
But now it seems I am getting somewhere.
I tried this:

```
root@VPN:~# docker run  -e WG_HOST='vpn.***********.com' ghcr.io/wg-easy/wg-easy
2024-11-29T20:04:19.891Z Server Listening on http://0.0.0.0:51821
2024-11-29T20:04:19.896Z WireGuard Loading configuration...
$ wg genkey
$ echo ***hidden*** | wg pubkey
2024-11-29T20:04:19.911Z WireGuard Configuration generated.
2024-11-29T20:04:19.911Z WireGuard Config saving...
$ wg-quick down wg0
2024-11-29T20:04:19.913Z WireGuard Config saved.
$ wg-quick up wg0
Error: WireGuard exited with the error: Cannot find device "wg0"
This usually means that your host's kernel does not support WireGuard!
    at /app/lib/WireGuard.js:74:17
    at async WireGuard.getConfig (/app/lib/WireGuard.js:72:7)
```

and this:
```

root@VPN:~# docker run  -e WG_HOST='vpn.**********.com' -i ghcr.io/wg-easy/wg-easy
2024-11-29T20:02:21.806Z Server Listening on http://0.0.0.0:51821
2024-11-29T20:02:21.810Z WireGuard Loading configuration...
$ wg genkey
$ echo ***hidden*** | wg pubkey
2024-11-29T20:02:21.823Z WireGuard Configuration generated.
2024-11-29T20:02:21.823Z WireGuard Config saving...
$ wg-quick down wg0
2024-11-29T20:02:21.826Z WireGuard Config saved.
$ wg-quick up wg0
Error: WireGuard exited with the error: Cannot find device "wg0"
This usually means that your host's kernel does not support WireGuard!
    at /app/lib/WireGuard.js:74:17
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async WireGuard.getConfig (/app/lib/WireGuard.js:72:7)
```

**Reply by shalafi99** [reply]
You are missing these:

`  --cap-add NET_ADMIN --cap-add SYS_MODULE --sysctl 'net.ipv4.conf.all.src_valid_mark=1' --sysctl 'net.ipv4.ip_forward=1'`

I tested it here, not including these above generated the error you saw.

But I'm puzzled, since you got a container going in the first place (the one for testing).
I'm not knowledgeable enough of the Linux network stack to understand how each parameter affects/impacts the container's behavior...

**Reply by Kid-Shelleen** [reply]
This is probably an old blog, but your host probably does not have wg0 but eth0.

**Comment by Kid-Shelleen** [comment]
Use eth0 instead of wg0. Some Linux distros (ie. Synology) use eth0.
